src/http.rs

```diff
@@ -26,17 +26,17 @@ pub fn retrieve_asset(
         } else {
             // url not in cache, we request it
             let mut response = client.get(url).send()?;
+            let res_url = response.url().to_string();
```

Comment:
`response.text()` now moves receiver. So accessing `response.url()` after `response.text()` is now complained by borrow checker. To avoid it, I bound the response URL here at first.